src/core/todoer.ts

```diff
@@ -1,6 +1,51 @@
-export function toggleTasks(todos: string): string {
-  const allLinesToggledOn = todos.replace(/^(?!- \[ \] )/gm, "- [ ] ");
-  return allLinesToggledOn === todos
-    ? todos.replace(/^- \[ \] /gm, "")
-    : allLinesToggledOn;
+/*
+^
+(?![ \t]*-[ \t]*\[[^\]]*\])    # Negative lookahead. Don't match lines that look like "- [<any_char>]"
+([ \t]*)                       # Capture leading indentation
+(?:-|\[[ \t]*\])?              # Non capture group. Match zero or one "-" or a "[ ]"
+[ \t]*(.*?)[ \t]*              # Capture the task text, leaving off any leading or trailing whitespace
+$
+*/
+const nonTasks = /^(?![ \t]*-[ \t]*\[[^\]]*\])([ \t]*)(?:-|\[[ \t]*\])?[ \t]*(.*?)[ \t]*$/gm;
+
+/*
+^
+(?!                                       # Negative lookahead.  Don't match completed or cancelled tasks
+  [ \t]*-[ \t]*\[[ \t]*x[ \t]*\]          # Don't match lines that start with "- [x]" (completed tasks)
+  |                                       # OR
+  [ \t]*-[ \t]*\[[ \t]*\][ \t]*~~[^~]+~~  # Don't match lines that match "- [ ] ~~<some text>~~" (cancelled tasks)
+)
+([ \t]*)                                  # Capture leading indentation 
+-[ \t]*\[[ \t]*\]                         # Match the a string that looks like "- [ ]"
+[ \t]*(.*?)[ \t]*                         # Capture the task text, leaving off any leading or trailing whitespace
+$
+*/
+const incompleteTasks =
+  /^(?![ \t]*-[ \t]*\[[ \t]*x[ \t]*\]|[ \t]*-[ \t]*\[[ \t]*\][ \t]*~~[^~]+~~)([ \t]*)-[ \t]*\[[ \t]*\][ \t]*(.*?)[ \t]*$/gm;
+
+/*
+^
+(?![ \t]*-[ \t]*\[[ \t]*\][ \t]*~~[^~]+~~) # Negative lookahead. Don't match lines that look like "- [ ] ~~<some text>~~" (cancelled tasks)
+([ \t]*)                                   # Capture leading indentation 
+-[ \t]*\[[ \t]*x[ \t]*\]                   # Match the a string that looks like "- [x]" (completed task)
+[ \t]*(.*?)[ \t]*                          # Capture the task text, leaving off any leading or trailing whitespace
+$
+*/
+const completeTasks = /^(?![ \t]*-[ \t]*\[[ \t]*\][ \t]*~~[^~]+~~)([ \t]*)-[ \t]*\[[ \t]*x[ \t]*\][ \t]*(.*?)[ \t]*$/gm;
```

Comment:
These regexes are so complex, and I'm totally open to simpler solution if one exists in the future.  I felt like I was on the "regex or nothgin!!!!" train.  Coudn't think of another way.  I'm curious to know how well they'll be maintainable.  

Anyways, they seem to work!